concert_software_farmer/src/concert_software_farmer/farmer.py

```diff
@@ -28,8 +28,21 @@ def __init__(self):
         self._software_pool = SoftwarePool()
         self._running_software = {}
         self._lock = threading.Lock()
+        self._timeout = rospy.get_param('~release_timeout', 0.15) #sec
+
 
         roslaunch.pmon._init_signal_handlers()
+        rospy.on_shutdown(self._wait_for_release_software)
+
+    def _wait_for_release_software(self):
+        '''
+          Wait for releasing the allocating software
+        '''
+        timeout_time = time.time() + self._timeout
+        while len(self._running_software) and time.time() < timeout_time:
+            rospy.sleep(0.2)
+        if len(self._running_software):
+            self.logwarn('Release timeout [%s]' % str(self._running_software.keys()))
```

Comment:
I developed printing warning log instead of raising timeout exception when ```release_timeout``` is over.
```_wait_for_release_software``` function is called at  ```roscore``` because software farmer registered it as shutdown callback, so we do not handle release timeout exception.